cpp/bench/prims/gram_matrix.cu

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/matrix/kernelparams.h>
+#include <matrix/grammatrix.h>
+#include <matrix/kernelfactory.h>
+#include <random/rng.h>
+#include <sstream>
+#include <string>
+#include <vector>
+#include "../common/ml_benchmark.hpp"
+
+namespace MLCommon {
+namespace Bench {
+namespace Matrix {
+
+using namespace MLCommon::Matrix;
+
+struct GramTestParams {
+  int m;  // m parameter of the GEMM
+  int k;  // k parameter of the GEMM
+  int n;  // n parameter of the GEMM
+  KernelParams kernel_params;
+};  // struct GramTestParams
+
+template <typename T>
+struct GramMatrix : public Fixture {
+  GramMatrix(const std::string& name, const GramTestParams& p)
+    : Fixture(name,
+              std::shared_ptr<deviceAllocator>(new defaultDeviceAllocator)),
+      params(p) {
+    std::vector<std::string> kernel_names{"linear", "poly", "rbf", "tanh"};
+    std::ostringstream oss;
+    oss << name << "/" << kernel_names[p.kernel_params.kernel] << "/" << p.m
+        << "x" << p.k << "x" << p.n;
+    this->SetName(oss.str().c_str());
+
+    CUBLAS_CHECK(cublasCreate(&cublas_handle));
+    kernel = Matrix::KernelFactory<T>::create(p.kernel_params, cublas_handle);
```

Comment:
this is not being `delete`d?

Reply:
It should be, fixed it. Thanks for catching it.